app/models/queries/sql_for_field.rb

```diff
@@ -119,6 +123,19 @@ def date_range_clause(table, field, from, to)
     s.join(' AND ')
   end
 
+  # Returns a SQL clause for date or datetime field for an exact range starting at the beginning of the day of from
+  # until the end of the day of to
+  def date_range_clause(table, field, from, to)
+    s = []
+    if from
+      s << ("#{table}.#{field} > '%s'" % [connection.quoted_date(from.to_time(form = :utc).beginning_of_day)])
+    end
+    if to
+      s << ("#{table}.#{field} <= '%s'" % [connection.quoted_date(to.to_time(form = :utc).end_of_day)])
```

Comment:
And might also be utterly wrong in case that the local timezone of the server is not UTC at all. Will remove that.